### PR TITLE
Autorepeatable keybindings & keybindings for label brush size

### LINKED ISSUE #3994: Brush size keybinding
## 🚀 Feature

Keybinding to change label paint brush size.
## Motivation

During labeling, I usually completely label each object (cell, cell colony, etc) before moving to another. To do so I label the coarse structure of the object first, using little zoom ("overview mode"). Then, I zoom in to label the finer details of the object (rim areas, branches, string-like features), before I switch to the next object. This involves quite a bit of paint brush size changing. 
Changing the size in the layer controls is inconvenient, as I have to move the cursor out of the image to be labeled, change the size, move back to the image to check if the new size is correct, (optionally move back to the layer controls to re-adjust), and then continue labeling. This disrupts the workflow.

## Pitch

I suggest a simple key binding (e.g. `q` and `e` to have them close to the `1`-`5` mode switches) to adjust the brush size.

## Alternatives

Alternatively, one could change the brush size using the mouse wheel (not used at the moment as far as I know).

## Additional context

This is my first request. I searched the issue list for potential duplicates of this request, but didn't find any, so I hope this request makes sense.
There already is a  menu `Preferences -> Shortcuts -> Group: Labels layer` to adjust keybinding, so I think it should be quite easy to add an option for brush size adjustment.

If anybody has pointers towards where this behavior is controlled, we ([MBPhys](https://github.com/MBPhys/) and me) are happy to implement this ourselves and merge it if you'll allow it.

### PR DESCRIPTION
**Pull request by RobAnKo**
# Description

As discussed in #3994, a keybinding to allow changing brush sizes in a Label layer is a helpful feature for manual labeling.
In addition, allowing the keybinding to autorepeat (hold the key to continuously increase/decrease the brush size) makes life even easier.
Upon implementing a hacky, hard-coded solution in PR #5024, we realized that a general pattern for autorepeatable keybindings would be a useful feature. Here I present a working solution.

Based on suggestions by @Czaki, 
- the Action class now has an additional field `repeatable : bool`, a flag that defines for each action whether it is autorepeatable.
- the ActionManager class has new methods `_is_repeatable` to check whether an action is repeatable, as well as `_get_repeatable_shortcuts` that returns a list of repeatable shortcuts. 
- At the moment the repeatability of actions is stored in a {name:bool} dictionary at napari/utils/repeatable_actions.py with the same keys as napari/utils/shortcuts.py
- In napari/utils/key_bindings  - on_key_press(), a call to action_manager._get_repeatable_shortcuts() is used to distinguish for a triggered action whether it is repeatable or not. 

Here you can see that press-and-hold works and that changing the hotkeys (from Q,E to Alt+A,Alt+D) in the settings does not break that functionality.
output_reduced

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update (not sure about this)

# References

based on discussions in #3994 and #5024, closes #5024, closes #3994 

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [x] example: all tests pass with my change
- [ ] example: I check if my changes works with both PySide and PyQt backends
      as there are small differences between the two Qt bindings.  

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/developers/translations.html).

### CODE REVIEW

File: napari/layers/labels/_labels_key_bindings.py
Comment (Czaki) [inline napari/layers/labels/_labels_key_bindings.py]: move to module level constraint

File: napari/layers/labels/_labels_key_bindings.py
Comment (Czaki) [inline napari/layers/labels/_labels_key_bindings.py]: move to module level containt

File: napari/utils/action_manager.py
Comment (Czaki) [inline napari/utils/action_manager.py]:
```suggestion
    repeatable: bool = False
```

File: napari/utils/action_manager.py
Comment (Czaki) [inline napari/utils/action_manager.py]:
```suggestion
        return self.actions[name].repeatable
```

File: napari/utils/action_manager.py
Comment (Czaki) [inline napari/utils/action_manager.py]:
```suggestion
            command, description, keymapprovider, repeatable
```

File: napari/utils/action_manager.py
Comment (Czaki) [inline napari/utils/action_manager.py]:
```suggestion
        active_func_names = set(i[1].__name__ for i in active_keymap.items())
```

File: napari/utils/action_manager.py
Comment (Czaki) [inline napari/utils/action_manager.py]:
```suggestion
                active_repeatable_shortcuts.extend(shortcuts)

```

File: napari/utils/key_bindings.py
Comment (Czaki) [inline napari/utils/key_bindings.py]:
```suggestion
        repeatables = {
            *action_manager._get_repeatable_shortcuts(self.keymap_chain),
            "Up",
            "Down",
            "Left",
            "Right",
        }
```

File: napari/utils/repeatable_actions.py
Comment (Czaki) [inline napari/utils/repeatable_actions.py]:
```suggestion
```

This should be set when registering actions

File: napari/utils/action_manager.py
Comment (Czaki) [inline napari/utils/action_manager.py]:
```suggestion
```

File: napari/layers/utils/layer_utils.py
Comment (Czaki) [inline napari/layers/utils/layer_utils.py]: Should be a passed argument of `register_layer_action` that needs to be added.

**Review by Czaki — COMMENTED** [review]
I'm happy to see that this thing is changing. Maybe even @andy-sweet will allow us to push it to `rc1`.  

If we create some structure to check if a member belongs to it, we should use `set` because of performance. 

I prefer to store information if an action is repeatable next to its definition, not in a separate structure.

**Comment by psobolewskiPhD** [comment]
From a functionality side, it works perfectly. Love it. 🥰

File: napari/layers/labels/_labels_key_bindings.py
Comment (RobAnKo) [reply napari/layers/labels/_labels_key_bindings.py]: Hey @Czaki, thanks for your suggestions, it is way cleaner this way. 
I'll implement them later today.

One question at this point: can you elaborate on what you mean with "move to module level constraint" here?

Comment (Czaki) [reply napari/layers/labels/_labels_key_bindings.py]: Create variables `MIN_BRUSH_SIZE = 1` and MAX_BRUSH_SIZE = 40` and compare to them, instead of hardcoded variables. 

This is halfway to making it fully configurable because someone could modify them from code.

File: napari/layers/labels/_labels_key_bindings.py
Comment (Czaki) [inline napari/layers/labels/_labels_key_bindings.py]:
```suggestion
@register_label_action(
    trans._("Decrease the paint brush size by one."), repetable=True
)
```

File: napari/layers/labels/_labels_key_bindings.py
Comment (Czaki) [inline napari/layers/labels/_labels_key_bindings.py]:
```suggestion
@register_label_action(
    trans._("Increase the paint brush size by one."), repetable=True
)
```

**Review by Czaki — APPROVED** [review]
Looks nice and ready to merge. Thanks for your contribution.

**Comment by Czaki** [comment]
@haesleinhuepf @psobolewskiPhD, are there any additions actions that you think should be repeatable?

**Comment by RobAnKo** [comment]
> Looks nice and ready to merge. Thanks for your contribution.

It was a pleasure, I learned a lot! Thank you for all the help along the way :+1:

**Comment by psobolewskiPhD** [comment]
> @haesleinhuepf @psobolewskiPhD, are there any additions actions that you think should be repeatable? 

Nothing else comes to my mind, other than movement keys (like arrows) and at present I don't think it's possible to move pts/shapes by key.

**Review by jni — APPROVED** [review]
Hahahaha that is a *badass* screengrab @RobAnKo! I was grinning ear-to-ear watching it. 😂 Thanks for everyone who has participated so far. Just want to lend my support to landing this in 0.4.17. 👍 🎉 🚀

File: napari/utils/action_manager.py
Comment (andy-sweet) [inline napari/utils/action_manager.py]:
```suggestion
            a boolean flag indicating whether the action can be autorepeated.
```

File: napari/utils/action_manager.py
Comment (andy-sweet) [inline napari/utils/action_manager.py]: Nit: remove the whitespace. Or maybe better, just condense the whole docstring into a one or two line summary as this is a private/helper function that is relatively simple.

Actually, is this used? I did a quick search for it on the GitHub file diff and did not find it. If that's correct, just delete this as the public equivalent is just as readable as calling this method (i.e. `self.actions[name].repeatable` vs.` self._is_repeatable(name)`).

**Comment by andy-sweet** [comment]
Thanks for the push here @RobAnKo ! As this already approved, I'll merge this after 24 hours so that it makes into v0.4.17.

I left a couple of nit-picky comments that would be nice to address before merging, but no problem if you don't get to it - we can follow up.

**Comment by RobAnKo** [comment]
Thanks for the nit-picks, I'll change them right away.

Good call on _is_repeatable, defined it during development and failed to realize it's not needed in the current implementation.

File: napari/utils/action_manager.py
Comment (RobAnKo) [reply napari/utils/action_manager.py]: Thanks for the nit-picks, I'll change them right away.

Good call on this private function, defined it during development and failed to realize it's not needed in the current implementation.

**Pull request merged**
